packages/plugins/apollo-usage-report/src/index.ts

```diff
@@ -271,3 +297,14 @@ async function sendTrace(
     logger.error('Failed to send trace:', err);
   }
 }
+
+function isDocumentNode(data: unknown): data is DocumentNode {
+  const isObject = (data: unknown): data is Record<string, unknown> =>
+    Object.prototype.toString.call(data).slice(8, -1) === 'Object';
```

Comment:
Is there a reason to use this trick instead of `typeof data === "object"` ?

Reply:
I'm not sure 🙂 , I just prefer `Object.prototype.toString.call(data).slice(8, -1) === 'Object'`. The only advantage in this case is that with this condition it will be really an object and it cannot be `null`. The equivalent condition with typeof would be `data && typeof data === 'object'`. I'll gladly change it to this condition. 🙂